### DISCUSSION TITLE
Stuck Installing/Uninstalling Homebrew. Please help

### DISCUSSION THREAD

**Discussion by mariella-infinitio** [Homebrew/discussions#444] [Getting started]
I’m completely stuck at installing and uninstalling Homebrew, I have zero experience on programming. My intention was to install FFmpeg on my Mac (Macos Big Sur Version 11.0) and failed at both after following instructions in its site and tutorial. Please, help me to either successfully install Homebrew to finally install FFmpeg , or to unistall it. I'm totally lost.

I followed several old threads to either reinstall or uninstall, and still no solution. I always get this error  **'Failed to locate Homebrew!'** Also when I try to install FFmpeg I get **'zsh: command not found: brew'**

I followed one thread that looks kind of similar to my situation. I followed this instructions at last (but I also tried other instructions lie I mention above): (https://github.com/Homebrew/homebrew-core/issues/5084)

cd /usr/local/Homebrew
git init
git remote add origin https://github.com/Homebrew/brew
git fetch --all
git checkout -B master origin/master
cd /usr/local
ln -s /usr/local/Homebrew/bin/brew /usr/local/bin

And I tried but this is what I get:

cd: no such file or directory: /usr/local/Homebrew
/usr/local/.git: Permission denied
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
ln: /usr/local/bin: Permission denied
➜  local 
 
Thank you! 
Mariella❤️

**Comment by SMillerDev** [comment]
Why not run the command on https://brew.sh/?

---
```
cd: no such file or directory: /usr/local/Homebrew
/usr/local/.git: Permission denied
```
This means you made `/usr/local` inaccessible to your user. You can run `sudo chown -R $(whoami) /usr/local/*` to restore the settings to the default. Afterwards I'd recommend following the suggestion I gave above.

**Reply by mariella-infinitio** [reply]
Thank you SMillerDev, I did as you suggested. In the end I always enter the command `brew help` to get started as they instruct but I keep getting 'zsh: command not found: brew' I get this answer thing...

➜  ~ git:(master) ✗ sudo chown -R $(whoami) /usr/local/*
Password:
➜  ~ git:(master) ✗


Warning: /opt/homebrew/bin is not in your PATH.
==> Installation successful!

==> Homebrew has enabled anonymous aggregate formulae and cask analytics.
Read the analytics documentation (and how to opt-out) here:
  https://docs.brew.sh/Analytics
No analytics data has been sent yet (or will be during this `install` run).

==> Homebrew is run entirely by unpaid volunteers. Please consider donating:
  https://github.com/Homebrew/brew#donations

==> Next steps:
- Run `brew help` to get started
- Further documentation: 
    https://docs.brew.sh
- Add Homebrew to your PATH in /Users/mariellalacunza/.zprofile:
    echo 'eval $(/opt/homebrew/bin/brew shellenv)' >> /Users/mariellalacunza/.zprofile
    eval $(/opt/homebrew/bin/brew shellenv)
➜  ~ git:(master) ✗ brew help
zsh: command not found: brew
➜  ~ git:(master) ✗ `brew help`
zsh: command not found: brew
➜  ~ git:(master) ✗ 

Help with that 'Add Homebrew to your PATH' instruction?? do I just run it in the terminal? which part? anyway I still get the error with brew help command...

**Reply by SMillerDev** [reply]
> Warning: /opt/homebrew/bin is not in your PATH.

It seems you're using an M1 powered macbook so most guides are not going to be accurate for you.

> Add Homebrew to your PATH in /Users/mariellalacunza/.zprofile:
```
echo 'eval $(/opt/homebrew/bin/brew shellenv)' >> /Users/mariellalacunza/.zprofile
eval $(/opt/homebrew/bin/brew shellenv)
```

You need to run these line in the output. Otherwise it won't work.

**Reply by foxt** [reply]
`This means you made /usr/local inaccessible to your user.`. This is how Apple intends `/usr/local` to be on modern Macs. (and cannot be changed `chown: /usr/local: Operation not permitted`)

There's a bug where installing brew on an Intel Mac will symlink `/usr/local` to `/opt/homebrew`, but won't be removed by the uninstaller. This will cause re-running the installer on an ASi machine that has been restored (i.e. via Time Machine or Migration Assistant) from that Intel Mac to try and install brew to `/opt/homebrew` which is an inaccessible folder.

```
# ls -l /opt
lrwxr-xr-x  1 foxt  11  2 Mar  2023 homebrew -> /usr/local/
```

To get rid, simply `sudo rm /opt/homebrew;` and try to install Homebrew again.

**Comment by mariella-infinitio** [comment]
Thank you Sean

Yes it's an M1 (I suspected that was the reason), I run that last command and I think it worked? it took about 7 min and I could run the FFmpeg but the app I want still doesn't recognize it.

It gave me a very long dialog, this is the last part of it. Do I need to run any of this to make my FFmpeg work? Do I need to ask for it in a separate discussion? Thank you :)

Dialog (sorry it's long)...

==> unbound
To have launchd start unbound now and restart at startup:
  sudo brew services start unbound
==> gnutls
If you are going to use the Guile bindings you will need to add the following
to your .bashrc or equivalent in order for Guile to find the TLS certificates
database:
  export GUILE_TLS_CERTIFICATE_DIRECTORY=/usr/local/etc/gnutls/
==> sqlite
sqlite is keg-only, which means it was not symlinked into /opt/homebrew,
because macOS already provides this software and installing another version in
parallel can cause all kinds of trouble.

If you need to have sqlite first in your PATH run:
  echo 'export PATH="/opt/homebrew/opt/sqlite/bin:$PATH"' >> ~/.zshrc

For compilers to find sqlite you may need to set:
  export LDFLAGS="-L/opt/homebrew/opt/sqlite/lib"
  export CPPFLAGS="-I/opt/homebrew/opt/sqlite/include"

For pkg-config to find sqlite you may need to set:
  export PKG_CONFIG_PATH="/opt/homebrew/opt/sqlite/lib/pkgconfig"

==> python@3.9
Python has been installed as
  /opt/homebrew/bin/python3

Unversioned symlinks `python`, `python-config`, `pip` etc. pointing to
`python3`, `python3-config`, `pip3` etc., respectively, have been installed into
  /opt/homebrew/opt/python@3.9/libexec/bin

You can install Python packages with
  pip3 install <package>
They will install into the site-package directory
  /opt/homebrew/lib/python3.9/site-packages

See: https://docs.brew.sh/Homebrew-and-Python
==> icu4c
icu4c is keg-only, which means it was not symlinked into /opt/homebrew,
because macOS provides libicucore.dylib (but nothing else).

If you need to have icu4c first in your PATH run:
  echo 'export PATH="/opt/homebrew/opt/icu4c/bin:$PATH"' >> ~/.zshrc
  echo 'export PATH="/opt/homebrew/opt/icu4c/sbin:$PATH"' >> ~/.zshrc

For compilers to find icu4c you may need to set:
  export LDFLAGS="-L/opt/homebrew/opt/icu4c/lib"
  export CPPFLAGS="-I/opt/homebrew/opt/icu4c/include"

For pkg-config to find icu4c you may need to set:
  export PKG_CONFIG_PATH="/opt/homebrew/opt/icu4c/lib/pkgconfig"

==> tesseract
This formula contains only the "eng", "osd", and "snum" language data files.
If you need any other supported languages, run `brew install tesseract-lang`.
➜  ~ git:(master) ✗

**Reply by SMillerDev** [reply]
> I could run the FFmpeg but the app I want still doesn't recognize it.

I suspect the app doesn't know how to deal with homebrew on M1 devices then. You should contact the developers to verify that.

> It gave me a very long dialog, this is the last part of it. Do I need to run any of this to make my FFmpeg work?

If you can run it, it's probably fine. If something is broken you can got to gist.github.com and put the complete log there, but I suspect that isn't necessary

**Comment by mariella-infinitio** [comment]
If it cannot be supported then, how can I uninstall Homebrew please to add ffmpeg the normal way?

**Comment by mariella-infinitio** [comment]
Update: The FFmpeg was correctly installed by Homebrew. It worked just fine. I'm working with Audacity for which there are custom FFmpeg codes, so Homebrew had nothing to do with it.
Thank you so much for your help, it's very appreciated.❤️
